backend/src/api/comment/create_comment.rs

```diff
@@ -120,4 +166,53 @@ mod test {
             assert_eq!(text, comment_text);
         });
     }
+
+    #[test_case("LMAO !pogu", "LMAO __pog__u__url__", vec![("pogu","__pog__u__url__")])]
+    #[test_case("LMAO!", "LMAO!", vec![("pogu","__pog__u__url__")])]
+    #[test_case("LMAO !pogu !clueless XDX!!!!!", "LMAO __pog__u__url__ __clueless_url__ XDX!!!!!", vec![("pogu","__pog__u__url__"), ("clueless", "__clueless_url__")])]
+    #[tokio::test]
+    async fn create_comment_alias(
+        comment_text: &str,
+        expected_comment_text: &str,
+        aliases: Vec<(&'static str, &'static str)>,
+    ) {
+        let state = AppState::in_memory_db().await;
+        let conn = state.pool.get().await.unwrap();
+
+        let result: anyhow::Result<(String, String, String)> = conn
+            .interact(move |conn| {
+                for alias in aliases {
+                    insert_alias(alias.0, alias.1, conn)?;
+                }
+
+                let user = insert_user("test", conn)?;
+                let image = insert_image(&user, conn).context("")?;
```

Comment:
Since this is a test feel free to just `unwrap()` errors, nice error handling is for the end user, I don't think it matters much in tests which are only seen by developers.
Either way you should not add an empty context to unwrap since the context should add more information to where the error happened.